FoldingTabBar/Controller/FoldingTabBarController/YALFoldingTabBarController.m

```diff
@@ -127,58 +106,52 @@ - (UIImage *)centerImageInTabBarView:(YALFoldingTabBar *)tabBarView {
+        [viewController tabBarWillCollapse:self.tabBarView];
     }
 }
 
-- (void)tabBarViewDidCollapse:(YALFoldingTabBar *)tabBarView {
-    id<YALTabBarInteracting>viewController = [self currentInteractingViewController];
-    if ([viewController respondsToSelector:@selector(tabBarViewDidCollapse)]) {
-        [viewController tabBarViewDidCollapse];
+-(void)tabBarDidCollapse:(YALFoldingTabBar *)tabBarView {
+    id<YALTabBarDelegate>viewController = [self currentInteractingViewController];
+    if ([viewController respondsToSelector:@selector(tabBarDidCollapse:)]) {
+        [viewController tabBarDidCollapse:self.tabBarView];
     }
 }
 
-- (void)tabBarViewWillExpand:(YALFoldingTabBar *)tabBarView {
-    id<YALTabBarInteracting>viewController = [self currentInteractingViewController];
-    if ([viewController respondsToSelector:@selector(tabBarViewWillExpand)]) {
-        [viewController tabBarViewWillExpand];
+-(void)tabBarWillExpand:(YALFoldingTabBar *)tabBarView {
+    id<YALTabBarDelegate>viewController = [self currentInteractingViewController];
+    if ([viewController respondsToSelector:@selector(tabBarWillExpand:)]) {
+        [viewController tabBarWillExpand:self.tabBarView];
     }
 }
 
-- (void)tabBarViewDidExpand:(YALFoldingTabBar *)tabBarView {
-    id<YALTabBarInteracting>viewController = [self currentInteractingViewController];
-    if ([viewController respondsToSelector:@selector(tabBarViewDidExpand)]) {
-        [viewController tabBarViewDidExpand];
+- (void)tabBarDidExpand:(YALFoldingTabBar *)tabBarView {
+    id<YALTabBarDelegate>viewController = [self currentInteractingViewController];
+    if ([viewController respondsToSelector:@selector(tabBarDidExpand:)]) {
+        [viewController tabBarDidExpand:self.tabBarView];
     }
 }
 
-- (void)extraLeftItemDidPressInTabBarView:(YALFoldingTabBar *)tabBarView {
-    id<YALTabBarInteracting>viewController = [self currentInteractingViewController];
-    if ([viewController respondsToSelector:@selector(extraLeftItemDidPress)]) {
-        [viewController extraLeftItemDidPress];
-    }
-}
+//TODO: fix
 
-- (void)extraRightItemDidPressInTabBarView:(YALFoldingTabBar *)tabBarView {
-    id<YALTabBarInteracting>viewController = [self currentInteractingViewController];
-    if ([viewController respondsToSelector:@selector(extraRightItemDidPress)]) {
-        [viewController extraRightItemDidPress];
+- (void)tabBarDidSelectExtraLeftItem:(YALFoldingTabBar *)tabBarView {
+    id<YALTabBarDelegate>viewController = [self currentInteractingViewController];
+    if ([viewController respondsToSelector:@selector(tabBarDidSelectExtraLeftItem:)]) {
+        [viewController tabBarDidSelectExtraLeftItem:self.tabBarView];
     }
 }
 
-- (void)itemInTabBarViewPressed:(YALFoldingTabBar *)tabBarView atIndex:(NSUInteger)index {
-    BOOL should = YES;
-    if ([self.delegate respondsToSelector:@selector(tabBarController:shouldSelectViewController:)]) {
```

Comment:
@mmrmmlrr in order to keep behaviour of UITabBarContoller we should manually call these methods of it's delegate
```
- tabBarController:shouldSelectViewController:
- tabBarController:didSelectViewController
```
Please call there methods when user tries to selects some item. Thanks.